utils/handles/socket.c

```diff
@@ -44,3 +46,135 @@ int open_client_socket(char const *const hostname, const size_t port) {
   // Return socket fd - 0 or greater
   return sockfd;
 }
+
+static void tls_init(mbedtls_ctx_t *tls_ctx) {
+  mbedtls_entropy_init(&tls_ctx->entropy);
+  mbedtls_ctr_drbg_init(&tls_ctx->ctr_drbg);
+  mbedtls_ssl_init(&tls_ctx->ssl);
+  mbedtls_ssl_config_init(&tls_ctx->ssl_conf);
+  mbedtls_x509_crt_init(&tls_ctx->cacert);
+  mbedtls_x509_crt_init(&tls_ctx->client_cacert);
+  mbedtls_pk_init(&tls_ctx->pk_ctx);
+  mbedtls_net_init(&tls_ctx->net_ctx);
+}
+
+static void tls_destroy(mbedtls_ctx_t *tls_ctx) {
+  mbedtls_entropy_free(&tls_ctx->entropy);
+  mbedtls_ctr_drbg_free(&tls_ctx->ctr_drbg);
+  mbedtls_ssl_free(&tls_ctx->ssl);
+  mbedtls_ssl_config_free(&tls_ctx->ssl_conf);
+  mbedtls_x509_crt_free(&tls_ctx->cacert);
+  mbedtls_x509_crt_free(&tls_ctx->client_cacert);
+  mbedtls_pk_free(&tls_ctx->pk_ctx);
+  mbedtls_net_free(&tls_ctx->net_ctx);
+}
+
+void tls_socket_close(mbedtls_ctx_t *tls_ctx) { tls_destroy(tls_ctx); }
+
+int tls_socket_send(mbedtls_ctx_t *ctx, char const *data, size_t size) {
+  return mbedtls_ssl_write(&ctx->ssl, (const unsigned char *)data, size);
+}
+
+int tls_socket_recv(mbedtls_ctx_t *ctx, char *data, size_t size,
+                    uint64_t timeout) {
+  return mbedtls_ssl_read(&ctx->ssl, (unsigned char *)data, size);
+}
+
+int tls_socket_connect(mbedtls_ctx_t *tls_ctx, char const *host, uint16_t port,
+                       char const *ca_pem, char const *client_cert_pem,
+                       char const *client_pk_pem, retcode_t *error) {
+  int mbedtls_ret = -1;
+  char const drgb_pres[] = "iota_tls_client";
+  bool is_client_auth = false;
+  char port_string[6];
+  sprintf(port_string, "%d", port);
```

Comment:
As `port_string` is not guaranteed to be 0-initialized, shouldn't we do something like `sprintf(port_string, "%06d", port);` ?